docs/src/testing/test-attributes.md

```diff
@@ -0,0 +1,135 @@
+## Test Attributes
+
+
+`snforge` allows setting some test attributes for test cases, in order to modify their behavior. 
+
+Currently, those attributes are supported:
+ 
+- `#[test]`
+- `#[ignore]`
+- `#[should_panic]`
+- `#[available_gas]`
+- `#[fork]`
+- `#[fuzzer]`
+
+### `#[test]`
+
+Marks the function as a test case, to be visible for test collector.
+Read more about test collection [here](./test-collection.md).
+
+### `#[ignore]`
+
+Marks the function as ignored, it will be skipped after collecting.
+Use this if you don't want the test to be run (the runner will display how many tests were ignored in the summary).
+
+Read more about the behavior and how to override this [here](./testing.md#ignoring-some-tests-unless-specifically-requested).
+
+### `#[should_panic]`
+
+A test function can be marked with this attribute, in order to assert that the test function itself will panic.
+If the test panics when marked with this attribute, it's considered as "passed".
+ 
+Moreover, it can be used with either a tuple of shortstrings or a string for assessment of the exit panic data 
+(depending on what your contract throws).
+
+#### Usage
+Asserting the panic data can be done with multiple types of inputs:
+
+Bytearray: 
+```rust
+#[should_panic(expected: "No such file or directory (os error 2)")]
+```
+
+Shortstring:
+```rust
+#[should_panic(expected: 'panic message')]
+```
+
+Array of shortstrings: 
+```rust
+#[should_panic(expected: ('panic message', 'second message',)]
+```
+
+Asserting that the function panics (any with any panic data): 
+
+```rust
+#[should_panic]
+```
+
+
+### `#[available_gas]`
+
+Sets a gas limit for the test.
+If the test exceeds the limit, it fails with an appropriate error. 
+
+#### Usage 
+
+Asserts that the test does not use more than 5 units of gas. 
+```rust
+#[available_gas(5)]
+```
+
+### `#[fork]`
+
+Enables state forking for the given test case.
+
+Read more about fork testing [here](../snforge-advanced-features/fork-testing.md).
+
+#### Usage
+
+Configures the fork endpoint with given URL/ID of a URL and a reference point for forking - block number, 
```

Comment:
What's `ID of a URL`? If you meant name of a fork config, then it is a bit confusing with the same info a few lines below